Add MQTT BBEs (#4829)
ballerina-platform/ballerina-distribution: examples/mqtt-service-ssl/mqtt_service_ssl.md

# MQTT service - SSL/TLS

> The `mqtt:Service` receives messages from the MQTT server using the `mqtt:Listener` via SSL/TLS. SSL/TLS can be enabled by configuring the `secureSocket`, which requires a certificate or a truststore. Further, mTLS can be enabled by providing a certificate and private key of the service or a keystore. Use this to connect to an MQTT server secured with SSL.

Reviewer: ```suggestion
The `mqtt:Service` receives messages from the MQTT server using the `mqtt:Listener` via SSL/TLS. SSL/TLS can be enabled by configuring the `secureSocket`, which requires a certificate or a trust store. Further, Mutual TLS (mTLS) can be enabled by providing a certificate and private key of the service or a keystore. This can be used to connect to an MQTT server secured with SSL.
```

::: code mqtt_service_ssl.bal :::

> ## Prerequisites

Reviewer: ```suggestion
## Prerequisites
```
Ditto

> - Start a [MQTT broker](https://mqtt.org/software/) instance configured to use SSL/TLS.

Reviewer: ```suggestion
- Start an [MQTT broker](https://mqtt.org/software/) instance, which is configured to use SSL/TLS.
```

> Run the program by executing the following command.

Reviewer: ```suggestion
Run the program by executing the following command.
```
Ditto

::: out mqtt_service_ssl.out :::

> >**Tip:** Run the MQTT client given in the [MQTT client - SSL/TLS](/learn/by-example/mqtt-client-ssl) example to publish some messages to the topic.

Reviewer: ```suggestion
>**Tip:** Run the MQTT client given in the [MQTT client - SSL/TLS](/learn/by-example/mqtt-client-ssl) example to publish some messages to the topic.
```
Should this be one of the steps (last step) rather than a tip? There should be a step if it is mandatory to do it when trying out the example.
